code/modules/cm_marines/vehicle_part_fabricator.dm

```diff
@@ -143,21 +151,24 @@
 				return
 			cost = initial(produce.point_cost)
 			build_part(produce, cost, user)
-			return
+			return TRUE
 
 	if(action == "cancel")
 		var/index = params["index"]
 
 		if(length(build_queue))
-			if(index == 1)
+			if(index > length(build_queue))
```

Comment:
This should still have a `null` check since @Drulikar was able to get `index = null` edge case
```suggestion
			if(index == null || index > length(build_queue))
```

Reply:
the error in drathek's screenshot is not null, it's going oob because the tgui value had not updated in the act. but sure